### PR TITLE
Fixed #30024 -- Made urlencode() and Client raise a TypeError when None is passed as data.

### PR DESCRIPTION
**Pull request by jdufresne**
https://code.djangoproject.com/ticket/30024

### CODE REVIEW

File: docs/ref/utils.txt [hunk 1]
```diff
@@ -700,6 +700,12 @@ escaping HTML.
     A version of Python's :func:`urllib.parse.urlencode` function that can
     operate on ``MultiValueDict`` and non-string values.
 
+    .. versionchanged:: 2.2
```
Comment (timgraham) [inline docs/ref/utils.txt]: Generally, we don't include versionchanged notes where no surrounding text is changed. If you don't think any text should remain after the annotation would be removed, then don't include the annotation. My opinion is that the annotation isn't required.

File: docs/releases/2.2.txt [hunk 2]
```diff
@@ -435,6 +435,16 @@ Miscellaneous
 * :attr:`Group.name <django.contrib.auth.models.Group.name>` ``max_length``
   is increased from 80 to 150 characters.
 
+* The :class:`django.test.Client` class now raises a ``TypeError`` if ``None``
```
Comment (timgraham) [inline docs/releases/2.2.txt]: I think the notes could be combined in some fashion to reduce redundancy. e.g. "Client / urlencode now raise TypeError if None is passed as a value of the data argument to the request methods / the query argument, respectively.

Maybe you want to try something a bit different.

File: tests/test_client/tests.py [hunk 3]
```diff
@@ -59,6 +59,15 @@ def test_query_string_encoding(self):
         response = self.client.get('/get_view/?var=1\ufffd')
         self.assertEqual(response.context['var'], '1\ufffd')
 
+    def test_get_data_none_raises(self):
+        get_data = {'value': None}
```
Comment (timgraham) [inline tests/test_client/tests.py]: variable could be omitted

**Comment by jdufresne** [comment]
Thanks for the review Tim. I've made the suggested changes. Let me know what you think about the edits to the release notes.

**Pull request merged**